alicloud/resource_alicloud_hbase_instance.go

```diff
@@ -0,0 +1,302 @@
+package alicloud
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/hbase"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func payTypePostPaidDiffSuppressFunc(k, old, new string, d *schema.ResourceData) bool {
+	return strings.ToLower(d.Get("pay_type").(string)) == "postpaid"
+}
+
+func resourceAlicloudHBaseInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudHBaseInstanceCreate,
+		Read:   resourceAlicloudHBaseInstanceRead,
+		Update: resourceAlicloudHBaseInstanceUpdate,
+		Delete: resourceAlicloudHBaseInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(2, 128),
+			},
+			"zone_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"engine": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"hbase", "hbaseue", "serverlesshbase", "spark", "bds"}, false),
+				Default:      "hbase",
+			},
+			"engine_version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"master_instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"core_instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"core_instance_quantity": {
+				Type:         schema.TypeInt,
+				ValidateFunc: validation.IntBetween(2, 20),
+				Optional:     true,
+				ForceNew:     true,
+				Default:      2,
+			},
+			"core_disk_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"cloud_ssd", "cloud_efficiency", "local_hdd_pro", "local_ssd_pro"}, false),
+			},
+			"core_disk_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(100, 2000),
+				Default:      100,
+			},
+			"pay_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{string(Postpaid), string(Prepaid)}, false),
```

Comment:
Please use `PostPaid` and `PrePaid` to replace `Postpaid` and `Prepaid`. If the api only support later, please convert them in the resource codes.